include/manager.h

```diff
@@ -44,25 +46,31 @@ void my_free(void* ptr);
  * Compartment function intercepts
  ******************************************************************************/
 
-// Intercept functions
-time_t manager_time();
-
 struct func_intercept {
     char* func_name;
     uintptr_t redirect_func;
-    void* __capability intercept_pcc;
     void* __capability intercept_ddc;
+    void* __capability intercept_pcc;
+    void* __capability redirect_cap;
 };
 
+void setup_intercepts();
+void print_full_cap(uintcap_t);
+void intercept_wrapper(void* to_call_fn);
+
+// Intercept functions
+time_t manager_time();
 
-static const struct func_intercept comp_intercept_funcs[] = {
+static const struct func_intercept to_intercept_funcs[] = {
     /* vDSO funcs */
     { "time", (uintptr_t) manager_time },
     //"printf",
     /* Mem funcs */
-    { "malloc", (uintptr_t) my_malloc },
-    { "realloc", (uintptr_t) my_realloc },
-    { "free", (uintptr_t) my_free },
+    //{ "malloc", (uintptr_t) my_malloc },
+    //{ "realloc", (uintptr_t) my_realloc },
+    //{ "free", (uintptr_t) my_free },
 };
+#define INTERCEPT_FUNC_COUNT sizeof(to_intercept_funcs) / sizeof(to_intercept_funcs[0])
```

Comment:
Should this go with the other `#define`s earlier in the file?

Reply:
It doesn't hurt to fix a few of the easy ones now and do a fuller pass later?

Reply:
I've done some reorganizing (https://github.com/capablevms/CHERI-ELF-comp/pull/2/commits/c97b5f75899653a517ef6824407457c6d756036f and https://github.com/capablevms/CHERI-ELF-comp/pull/2/commits/e6c68bf0986f162137101dc35c7a2d42f6e6b11a). I still kept the `#define`s separate, as I split everything by category, and it felt more natural like so. I could put all `#define`s together on top, if you think it's best, but `INTERCEPT_FUNC_COUNT` will have to be separate, as it's dynamically computed (but needs to be `#define`d as it's used as an array argument).

Reply:
I forgot a few small additions in another file (removing an obsolete `#define` use and adding the implementation of `fprintf`) and added them in https://github.com/capablevms/CHERI-ELF-comp/pull/2/commits/d919632f88efccb19080646b089771965c006726.

Reply:
Works for me.